cilium/cmd/fqdn.go

```diff
@@ -43,31 +50,83 @@ var fqdnCleanCacheCmd = &cobra.Command{
 	},
 }
 
-var fqdnCacheClearMatchPattern string
+var fqdnListCacheCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List fqdn cache contents",
+	Run: func(cmd *cobra.Command, args []string) {
+		listFQDNCache()
+	},
+}
+
+var fqdnCacheMatchPattern string
 
 func init() {
+	fqdnCacheCmd.AddCommand(fqdnListCacheCmd)
 	fqdnCacheCmd.AddCommand(fqdnCleanCacheCmd)
 	fqdnCmd.AddCommand(fqdnCacheCmd)
 	rootCmd.AddCommand(fqdnCmd)
 
 	fqdnCleanCacheCmd.Flags().BoolVarP(&force, "force", "f", false, "Skip confirmation")
-	fqdnCleanCacheCmd.Flags().StringVarP(&fqdnCacheClearMatchPattern, "pattern", "p", "", "Cache entries with fqdn matching the pattern will be deleted")
+	fqdnCleanCacheCmd.Flags().StringVarP(&fqdnCacheMatchPattern, "pattern", "p", "", "Cache entries with fqdn matching the pattern will be deleted")
+
+	fqdnListCacheCmd.Flags().StringVarP(&fqdnCacheMatchPattern, "pattern", "p", "", "List cache entries with FQDN matching the pattern")
+	command.AddJSONOutput(fqdnListCacheCmd)
 }
 
 func cleanFQDNCache() {
-	if !force && !confirmCleanup() {
-		return
+	if !force {
+		fmt.Println("Following cache entries are going to be deleted:")
+		listFQDNCache()
+		if !confirmCleanup() {
+			return
+		}
 	}
 
 	params := policy.NewDeleteFqdnCacheParams()
 
-	if fqdnCacheClearMatchPattern != "" {
-		params.SetMatchpattern(&fqdnCacheClearMatchPattern)
+	if fqdnCacheMatchPattern != "" {
+		params.SetMatchpattern(&fqdnCacheMatchPattern)
 	}
 
 	_, err := client.Policy.DeleteFqdnCache(params)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error: %s\n", err)
+		return
```

Comment:
This should be an `os.Exit(1)`, shouldn't it? (since it is an error condition). We also would need to update the error return in listFQDNCache.

Reply:
Actually, since we have a `Fatalf` from somewhere, these should all be `Fatalf` calls instead.

Reply:
done